espnet2/enh/separator/dptnet_separator.py

```diff
@@ -0,0 +1,189 @@
+            num_spk: number of speakers
+            unit: int, dimension of the hidden state.
+            att_heads: number of attention heads.
+            dropout: float, dropout ratio. Default is 0.
+            activation: activation function applied at the output of RNN.
+            norm_type: type of normalization to use after each inter- or
+                intra-chunk Transformer block.
+            nonlinear: the nonlinear function for mask estimation,
+                       select from 'relu', 'tanh', 'sigmoid'
+            layer: int, number of stacked RNN layers. Default is 3.
+            segment_size: dual-path segment size
+        """
+        super().__init__()
+
+        self._num_spk = num_spk
+        self.feature_dim = feature_dim
+        self.segment_size = segment_size
+
+        self.enc_LN = choose_norm(norm_type, input_dim)
+        self.dptnet = DPTNet(
+            rnn_type=rnn_type,
+            input_size=input_dim,
+            hidden_size=unit,
+            output_size=input_dim * num_spk,
+            att_heads=att_heads,
+            dropout=dropout,
+            activation=activation,
+            num_layers=layer,
+            bidirectional=bidirectional,
+            norm_type=norm_type,
+        )
+        # gated output layer
+        self.output = torch.nn.Sequential(
+            torch.nn.Conv1d(input_dim, input_dim, 1), torch.nn.Tanh()
+        )
+        self.output_gate = torch.nn.Sequential(
+            torch.nn.Conv1d(input_dim, input_dim, 1), torch.nn.Sigmoid()
+        )
+
+        if nonlinear not in ("sigmoid", "relu", "tanh"):
+            raise ValueError("Not supporting nonlinear={}".format(nonlinear))
+
+        self.nonlinear = {
+            "sigmoid": torch.nn.Sigmoid(),
+            "relu": torch.nn.ReLU(),
+            "tanh": torch.nn.Tanh(),
+        }[nonlinear]
+
+    def forward(
+        self,
+        input: Union[torch.Tensor, ComplexTensor],
+        ilens: torch.Tensor,
+        additional: Optional[Dict] = None,
+    ) -> Tuple[List[Union[torch.Tensor, ComplexTensor]], torch.Tensor, OrderedDict]:
+        """Forward.
+
+        Args:
+            input (torch.Tensor or ComplexTensor): Encoded feature [B, T, N]
+            ilens (torch.Tensor): input lengths [Batch]
+            additional (Dict or None): other data included in model
+                NOTE: not used in this model
+
+        Returns:
+            masked (List[Union(torch.Tensor, ComplexTensor)]): [(B, T, N), ...]
+            ilens (torch.Tensor): (B,)
+            others predicted data, e.g. masks: OrderedDict[
+                'mask_spk1': torch.Tensor(Batch, Frames, Freq),
+                'mask_spk2': torch.Tensor(Batch, Frames, Freq),
+                ...
+                'mask_spkn': torch.Tensor(Batch, Frames, Freq),
+            ]
+        """
+
+        # if complex spectrum,
+        if is_complex(input):
+            feature = torch.nn.functional.relu(abs(input))
```

Comment:
maybe leave relu optional ?

Reply:
I notice that it is not necessary to apply ReLU to `abs(input)`.
For the `else` branch, I will make ReLU optional.